File: backend/app/schemas/supplier.py
from typing import Optional, Literal

from pydantic import BaseModel, EmailStr, Field


PaymentStatus = Literal["paid", "pending", "cancelled"]


class SupplierCreate(BaseModel):
    name: str
    company: str = ""
    email: EmailStr
    phone: str
    address: str

    totalPurchase: float = Field(default=0, ge=0)
    pendingPayment: float = Field(default=0, ge=0)

    paymentStatus: PaymentStatus = "paid"


class SupplierUpdate(BaseModel):
    name: Optional[str] = None
    company: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    address: Optional[str] = None

    totalPurchase: Optional[float] = Field(
        default=None,
        ge=0
    )

    pendingPayment: Optional[float] = Field(
        default=None,
        ge=0
    )

    paymentStatus: Optional[PaymentStatus] = None

File: frontend/src/types/supplier.ts
export type PaymentStatus =
  | "paid"
  | "pending"
  | "cancelled";

export interface Supplier {
  id: string | number;
  name: string;
  company: string;
  phone: string;
  email: string;
  address: string;
  totalPurchase: number;
  pendingPayment: number;
  paymentStatus: PaymentStatus;
}